File: php/features.php
<?php
declare(strict_types=1);

// PublicTime SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PublicTimeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PublicTimeBaseFeature();
            case "test":
                return new PublicTimeTestFeature();
            default:
                return new PublicTimeBaseFeature();
        }
    }
}

File: py/features.py
# PublicTime SDK feature factory

from feature.base_feature import PublicTimeBaseFeature
from feature.test_feature import PublicTimeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PublicTimeBaseFeature(),
        "test": lambda: PublicTimeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
